packages/components/src/CONTRIBUTING.md

```diff
@@ -0,0 +1,80 @@
+4. **Provide usage documentation**: If the contribution adds additional behavior or expands a component’s features, we’ll need to document them with development and design guidelines. Read through existing component documentation for examples.
+5. **Provide working code**: The component or enhancement must be built in React. See the [developer contribution guidelines](https://github.com/WordPress/gutenberg/blob/master/docs/contributors/develop.md).
+
+Remember, it’s unlikely that all parts will be done by one person. Someone in the community may be able to help.
+
+## Component refinement
+
+Before a component is published it will need to be fine-tuned:
+
+1. **Expand** to a minimally sufficient feature set. Resolve what features should be included in the component.
+2. **Reduce** scope to omit contentious features. Review proposed features and prune unessential items lacking consensus.
+3. **Quality assurance**: each contribution must be normalized into system standards.
+
+### Quality assurance
+
+To ensure quality, each component should be tested. The testing process should be done during the development of the component and before the component is published. 
+
+- **Accessibility**: Has the design and implementation accounted for accessibility? Please use the [WordPress accessibility guidelines](https://make.wordpress.org/accessibility/handbook/best-practices/).
+- **Visual quality**: Does the component apply visual style — color, typography, icons, space, borders, and more — using appropriate variables, and does it follow [visual guidelines](https://make.wordpress.org/design/handbook/design-guide/)?
+- **Responsiveness**: Does it incorporate responsive display patterns and behaviors as needed? Is the component designed from a mobile-first perspective? Do all touch interactions work as expected?
+- **Sufficient states & variations**: Does it cover all the necessary variations (primary, secondary, dense, etc.) and states (default, hover, active, disabled, loading, etc.), given the intended scope?
+- **Content resilience**: Is each dynamic word or image element resilient to too much, too little, and no content at all, respectively? How long can labels be, and what happens when you run out of space?
+- **Composability**: Does it fit well when placed next to or layered with other components to form a larger composition?
+- **Functionality**: Do all behaviors function as expected?
+- **Browser support**: Has the component visual quality and accuracy been assessed across Safari, Chrome, Firefox, IE, and other browsers across relevant devices? Please adhere to our [browser support requirements](https://github.com/WordPress/gutenberg/blob/master/packages/browserslist-config/index.js).
+- **Documentation**: Ensure that the component has proper documentation for development, design, and accessibility.
+
+## Deprecation
+
+A component/feature may need deprecation if:
```

Comment:
We should emphasize that this applies not only to components but also to their props, whenever we add a new one, rename or remove. This should be reflected in the docs but at the same time we should ensure that old code still works as close as possible to what was before and deprecation warnings are outputted to the Web Console.

Reply:
I wonder if this section belongs to this document, since this is generic and applies to any public API of the project. (A backward compatibility policy document)

Reply:
From my experience, it doesn't harm to repeat it a couple of times in our docs :D

Reply:
I have no idea where our backward compatibility policy document is—linking to it here might be a good approach, but it's definitely helpful being as explicit as possible about our approach here, specifically with reference to how that works for components.

Reply:
I agree there are some component specific things that can be included here.